python/plugins/processing/algs/qgis/CreateAtlasGrid.py

```diff
@@ -0,0 +1,178 @@
+
+    def displayName(self):
+        # The name that the user will see in the toolbox
+        return self.tr('Create a grid from map layout')
+
+    def group(self):
+        return self.tr('Cartography')
+
+    def groupId(self):
+        return 'cartography'
+
+    def __init__(self):
+        super().__init__()
+
+    def initAlgorithm(self, config=None):
+
+        layout_param = QgsProcessingParameterString(
+            self.LAYOUT,
+            description=self.tr('Layout to use'),
+            defaultValue=None,
+            optional=False)
+
+        layout_param.setMetadata(
+            {'widget_wrapper': {
+                'class':
+                    'processing.gui.wrappers_layout.LayoutWrapper'}})
+        self.addParameter(layout_param)
+
+        self.addParameter(QgsProcessingParameterFeatureSource(self.COVERAGE_LAYER,
+                                                              self.tr('Coverage layer'),
+                                                              [QgsProcessing.TypeMapLayer]))
+        self.addParameter(QgsProcessingParameterDistance(self.COVERAGE_BUFFER,
+                                                         self.tr('Coverage Buffer'), parentParameterName='COVERAGE_LAYER',
+                                                         defaultValue=0.0))
+        self.addParameter(QgsProcessingParameterBoolean(self.ONLY_ON_FEATURES,
+                                                        self.tr('Only on intersection features'),
+                                                        defaultValue=True))
+        self.addParameter(QgsProcessingParameterFeatureSink(self.OUTPUT,
+                                                            self.tr('Grid'),
+                                                            QgsProcessing.TypeVectorPolygon))
+
+    def processAlgorithm(self, parameters, context, feedback):
+
+        layout_name = self.parameterAsString(
+            parameters,
+            self.LAYOUT,
+            context
+        )
+        coverage_layer = self.parameterAsVectorLayer(
+            parameters,
+            self.COVERAGE_LAYER,
+            context
+        )
+        coverage_distance = self.parameterAsDouble(
+            parameters,
+            self.COVERAGE_BUFFER,
+            context
+        )
+        only_on_features = self.parameterAsBool(
+            parameters,
+            self.ONLY_ON_FEATURES,
+            context
+        )
+
+        fields = QgsFields()
+        fields.append(QgsField('id', QVariant.Int, '', 10, 0))
+        (sink, dest_id) = self.parameterAsSink(
+            parameters,
+            self.OUTPUT,
+            context,
+            fields,
+            QgsWkbTypes.Polygon,
+            coverage_layer.sourceCrs()
+        )
+
+        if sink is None:
+            raise QgsProcessingException(self.invalidSinkError(parameters, self.OUTPUT))
+
+        layout = QgsProject().instance().layoutManager().layoutByName(layout_name)
```

Comment:
if layout is not available => error that is not managed in this code... if you use Enum you can avoid the user to use other vaulaes than that listed in available layouts

Reply:
I think this will never happen, but if you consider it, I can control the possible error.

Reply:
Actually it's entirely possible -- algorithms can be run from code and accordingly parameter values can be set to any possible value, not just those which are exposed when the algorithm is run from the gui.